src/nvme/ioctl.c

```diff
@@ -1670,42 +1670,40 @@ static int nvme_set_var_size_tags(__u32 *cmd_dw2, __u32 *cmd_dw3, __u32 *cmd_dw1
 		__u8 pif, __u8 sts, __u64 reftag, __u64 storage_tag)
 {
 	__u32 cdw2 = 0, cdw3 = 0, cdw14;
-	beint64_t be_reftag = cpu_to_be64(reftag);
-	beint64_t be_storage_tag = cpu_to_be64(storage_tag);
 
 	switch (pif) {
 	case NVME_NVM_PIF_16B_GUARD:
-		cdw14 = be_reftag & 0xffffffff;
-		cdw14 |= ((be_storage_tag << (32 - sts)) & 0xffffffff);
+		cdw14 = (reftag >> 32) & 0xffffffff;
```

Comment:
The shift here would still reduce any valid 32-bit tag to 0, but otherwise with that removed the updated changes look good to me.

```suggestion
		cdw14 = reftag & 0xffffffff;
```

Reply:
Thanks for pointing that out, PR is updated!